### ISSUE TITLE
Compatiblity with Socket 2.x

### ISSUE THREAD

**Issue by srinisubramanian** [socketio/socket.io-emitter#63]
After upgrading Socket to 2.x (earlier we were on 1.x), the emitted events/messages are not reaching the clients.  socket.io-emitter used is the latest release.

"socket.io": "2.0.3"
"socket.io-client": "2.0.3"
"socket.io-emitter": "3.1.0"
"socket.io-redis": "5.1.0"

Is the current version confirmed to be compatible/work with Socket 2.x ?  Or any adjustments are needed in code?

Thanks

**Comment by darrachequesne** [comment]
`"socket.io-emitter": "3.1.0"` should indeed be compatible with `"socket.io-redis": "5.1.0"`, which in turn is compatible with `"socket.io-client": "2.0.3"`

Are you using a custom redis client?

**Comment by srinisubramanian** [comment]
We are currently using  "redis": "2.7.1".  One additional observation: When the socket server is restarted the event is being posted.  Looks like maybe there is something happening in the subscription side of things?

**Comment by srinisubramanian** [comment]
Additionally: Downgrading our app to redis 2.6.3 got everything working again.  Issue seems to be socket.io-emitter with redis 2.7.1

**Comment by vinodjoseph62** [comment]
The messages are reaching the clients after upgrading socket-io-emitter's redis dependency in package json to 2.7.1 (earlier it was 2.6.3) .

**Comment by srinisubramanian** [comment]
@darrachequesne Any ideas when you will get an opportunity to look at this?  We have upgraded already and the emitter is not working and hence had to downgrade.  If you need any additional information please do let me know.

**Comment by srinisubramanian** [comment]
@darrachequesne Apologies.  It turned out to be an issue at our end.  When we start socket.io-redis, we do not pass in credentials but pass in the actual pub/sub client connection.  We had set return_buffers=true earlier (since we also used the connection) and this was causing the issue.  Setting return_buffers=false and passing it got everything working.

Thanks for your help, I am closing the issue as this is not a problem.
